packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -183,11 +204,53 @@ export class HttpRoute extends Resource implements IHttpRoute {
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
       authorizerId: authBindResult?.authorizerId,
-      authorizationType: authBindResult?.authorizationType ?? 'NONE', // must be explicitly NONE (not undefined) for stack updates to work correctly
+      authorizationType: Lazy.string({
+        produce: () => {
+          if (this.iamEnabled) {
+            return 'AWS_IAM';
+          } else {
+            return authBindResult?.authorizationType ?? 'NONE'; // must be explicitly NONE (not undefined) for stack updates to work correctly
+          }
```

Comment:
(minor) else block is unneeded.

```suggestion
          return authBindResult?.authorizationType ?? 'NONE'; // must be explicitly NONE (not undefined) for stack updates to work correctly
```

Reply:
Done.